apps/server/src/core/error/filter/global-error.filter.ts

```diff
@@ -114,15 +116,21 @@ const writeErrorLog = (error: unknown, logger: ILogger): void => {
 };
 
 @Catch()
-export class GlobalErrorFilter<T = unknown> implements ExceptionFilter<T> {
+export class GlobalErrorFilter<T extends { traceId?: string }> implements ExceptionFilter<T> {
 	constructor(private logger: Logger) {
 		this.logger.setContext('Error');
 	}
 
-	// eslint-disable-next-line class-methods-use-this
 	catch(error: T, host: ArgumentsHost): void {
 		const ctx = host.switchToHttp();
+		const request = ctx.getRequest<Request>();
 		const response = ctx.getResponse<Response>();
+		const cookies = cookie.parse(request.headers.cookie || '');
+		const traceId: string = cookies.traceId || uuid();
```

Comment:
doesnt this assign the traceId to the user instead of the request?
Each request of the same user should have its own traceid in my opinion.